src/Controller/UserController.php

```diff
@@ -764,8 +796,36 @@ public function setProjectManagerForUser(
                 'message' => 'Access denied!'
             ], Response::HTTP_FORBIDDEN);
         }
+
+        $data = $request->getContent();
+        /** @var DeserializationContext $context */
+        $context = DeserializationContext::create()->setGroups(array('SetRole'));
+
+        $id = $this->serializer->deserialize(
+            $data,
+            User::class,
+            'json',
+            $context
+        );
+        $projectManager = $userRepository->find($id);
+        if (!$projectManager) {
+            return new JsonResponse(
+                [
+                    'code' => Response::HTTP_NOT_FOUND,
+                    'message' => 'Not found!'
+                ],
+                Response::HTTP_NOT_FOUND
+            );
+        }
+
         if ($projectManager->getRoles() === array('ROLE_USER')) {
```

Comment:
This isn't how you validate if a user has a specific role. Also, you should check specificaly for ROLE_PM as there could be roles like DEV, QA and others.